rfcs/0001-reason-message-diff.md

```diff
@@ -0,0 +1,53 @@
+### Context:
+
+When kpack produces new builds today, a Reason is given which may be one of five values: COMMIT, TRIGGER, CONFIG, STACK, BUILDPACK
+
+These one-word values are helpful for providing a basic level of context on why a new build has been created, but they don't provide enough information to inform users' decisions about what to do with the resulting build. They may be asking themselves: "Does this new build have some update that affects my app? Should I run my regression tests on it? Should I promote it to prod?" 
```

Comment:
It seems like we are targeting two different use cases here. (Why is a build kicked off & what is different about a resulting image). 

The "Why is a build kicked off" question is one that seems like it could be solved by expanding the BuildReason. However, the "what is different about a resulting image" seems like it could be better solved disconnected from the BuildReason.

Reply:
> However, the "what is different about a resulting image" seems like it could be better solved disconnected from the BuildReason.

A specific situation I am thinking about for this is when a new COMMIT build leads to the change in a dependency version provided by a buildpack such as the JVM.

Reply:
I'm very interested in a `kp image diff <build-A> <build-B>` command and I'm wondering if that is a useful place for this feature. Do we think users have a desire to diff any two builds for an image? The default could simply be the two most recent builds.

Reply:
I think this interface could be an elegant way to handle both build diff and more details on the reason, but there is a risk here around edge cases either bloating the interface or causing it to break or not be useful in certain cases.  To me the next step is to figure out if it is sensible to handle both of these use cases together.  If no, then maybe focus on the status first and submit a different RFC that presents an interface to handle build diffs.

Reply:
@djoyahoy I think one of the reasons why I find the UX of this RFC attractive is that the user wouldn't have to specify the builds they want to compare against.  The metadata about what changed from the last build would be available on every build.  Preserving that UX I think would be ideal rather than juggling build numbers.